src/Scaffolding/VS.Web.CG.Mvc/BlazorIdentity/BlazorIdentityGenerator.cs

```diff
@@ -0,0 +1,544 @@
+            await ModifyProgramCsAsync(programCsFile, blazorIdentityModel);
+
+            //Modify all other non .cs files
+            var otherFiles = minimalApiChangesConfig.Files.Where(x => !x.FileName.EndsWith(".cs", StringComparison.OrdinalIgnoreCase));
+            var projectPath = Path.GetDirectoryName(ProjectContext.ProjectFullPath);
+            var project = Workspace.CurrentSolution.Projects.FirstOrDefault(p => p.AssemblyName.Equals(ProjectContext.AssemblyName, StringComparison.OrdinalIgnoreCase));
+            foreach (var file in otherFiles)
+            {
+                var fileDoc = project.GetDocumentFromName(file.FileName, FileSystem);
+                await DocumentBuilder.ApplyTextReplacements(file, fileDoc, new CodeChangeOptions());
+                if (file.FileName.Equals("Routes.razor", StringComparison.OrdinalIgnoreCase) &&
+                    FileSystem.FileExists(fileDoc.Name))
+                {
+                    var fileText = FileSystem.ReadAllText(fileDoc.Name);
+                    fileText = $"@using {blazorIdentityModel.BlazorIdentityNamespace}.Shared" + Environment.NewLine + fileText;
+                    FileSystem.WriteAllText(fileDoc.Name, fileText.ToString());
+                    ConsoleLogger.LogMessage($"Modified {fileDoc.Name}.\n");
+                }
+            }
+        }
+
+        internal async Task ModifyProgramCsAsync(CodeFile programCsFile, BlazorIdentityModel blazorIdentityModel)
+        {
+            if (programCsFile is null)
+            {
+                return;
+            }
+
+            var programType = ModelTypesLocator.GetType("<Program>$").FirstOrDefault() ?? ModelTypesLocator.GetType("Program").FirstOrDefault();
+            var project = Workspace.CurrentSolution.Projects.FirstOrDefault(p => p.AssemblyName.Equals(ProjectContext.AssemblyName, StringComparison.OrdinalIgnoreCase));
+            var programDocument = project.GetUpdatedDocument(FileSystem, programType);
+            //Modifying Program.cs document
+            var docEditor = await DocumentEditor.CreateAsync(programDocument);
+            var docRoot = docEditor.OriginalRoot as CompilationUnitSyntax;
+            programCsFile.Usings = programCsFile.Usings.Union(new string[] { blazorIdentityModel.DbContextNamespace, blazorIdentityModel.BlazorIdentityNamespace }).ToArray();
+            var docBuilder = new DocumentBuilder(docEditor, programCsFile, ConsoleLogger);
+            var newRoot = docBuilder.AddUsings(new CodeChangeOptions());
+            var useTopLevelsStatements = await ProjectModifierHelper.IsUsingTopLevelStatements(project.Documents.ToList());
+            var builderIdentifier = ProjectModifierHelper.GetBuilderVariableIdentifierTransformation(newRoot.Members);
+            //add code snippets/changes.
+            if (programCsFile.Methods != null && programCsFile.Methods.Count != 0)
```

Comment:
`if (programCsFile.Methods?.Count > 0)` ?

Reply:
yep that's better tyty.